server/tests/it/test_api_tournaments.py

```diff
@@ -0,0 +1,175 @@
+from datetime import datetime
+from typing import Tuple
+import uuid
+
+from QRServer.db.models import DbMatchReport
+from . import QuadradiusIntegrationTestCase
+
+
+class ApiTournamentsIT(QuadradiusIntegrationTestCase):
+    async def itSetUpConfig(self, config):
+        config.set('api.enabled', True)
+        config.set('auth.auto_register', True)
+
+    async def _make_two_users(self) -> Tuple[str, str]:
+        """
+        Returns:
+            (str, str): tuple of both user ids
+        """
+        client1 = await self.new_lobby_client()
+        await client1.join_lobby('PlayerA', 'ff585d509bf09ce1d2ff5d4226b7dacb')
+        user1 = await self.connector.get_user_by_username('PlayerA')
+        if not user1:
+            raise ValueError('Failed to create user 1')
+
+        client2 = await self.new_lobby_client()
+        await client2.join_lobby('PlayerB', 'ff585d509bf09ce1d2ff5d4226b7dacb')
+        user2 = await self.connector.get_user_by_username('PlayerB')
+        if not user2:
+            raise ValueError('Failed to create user 2')
+
+        return (user1.user_id, user2.user_id)
+
+    async def test_list_tournament_users(self):
+        tournament_id = await self.connector.create_tournament('test_name', '123', '234', 3)
+
+        user1_id, user2_id = await self._make_two_users()
+        result = await self.connector.add_participant(tournament_id, user1_id)
+        self.assertTrue(result)
+        result = await self.connector.add_participant(tournament_id, user2_id)
+        self.assertTrue(result)
+
+        api_client = await self.new_api_client('v1')
+        async with api_client.get(f'tournaments/{tournament_id}/users') as r:
+            self.assertEqual(r.status, 200)
+            self.assertEqual(await r.json(), {
+                user1_id: {
+                    'id': user1_id,
+                    'username': 'PlayerA',
```

Comment:
It's not obvious why you put `PlayerA` in here. In fact, this refers to an implementation detail of `self._make_two_users`. Either:
1. you pass this data to `_make_two_users` (i.e. add arguments), or
2. you get this data from `_make_two_users` (i.e. return the user, not its ID).

Reply:
done, I went with 2